include/forcing/CsvPerFeatureForcingProvider.hpp

```diff
@@ -143,7 +145,15 @@ class CsvPerFeatureForcingProvider : public forcing::ForcingProvider
             else
                 value += involved_time_step_values[i] * ((double)involved_time_step_seconds[i] / (double)duration_s);
         }
-        return value;
+
+        // Convert units
+        try {
+            return UnitsHelper::get_converted_value(available_forcings_units[output_name], value, output_units);
+        }
+        catch (const std::runtime_error& e){
+            std::cerr<<"Unit conversion error: "<<std::endl<<e.what()<<std::endl<<"Returning unconverted value!"<<std::endl;
```

Comment:
Yes, a custom exception subclass might make sense here also. But concur, later...it will be interesting to see how often this occurs once we start running it.